actionpack/lib/action_dispatch/system_test_case.rb

```diff
@@ -84,9 +84,29 @@ class SystemTestCase < IntegrationTest
     include SystemTesting::TestHelpers::SetupAndTeardown
     include SystemTesting::TestHelpers::ScreenshotHelper
 
+    class TeardownGatekeeper # :nodoc:
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
+        ActionDispatch::SystemTestCase.lock.synchronize do
+          if ActionDispatch::SystemTestCase.running?
+            @app.call(env)
+          else
+            [200, { "Content-Type" => "text/plain" }, ["OK"]]
```

Comment:
I wonder if Rack has a way to just drop the connection? Or failing that, maybe a 500 status would be better. It feels a little weird to return a fake success status.